lib/checkunusedvar.cpp

```diff
@@ -1658,6 +1658,20 @@ bool CheckUnusedVar::isFunctionWithoutSideEffects(const Function& func, const To
                         pointersToGlobals.insert(assigned_var_token->variable());
                     }
                 }
+                // dereference & assignment to unknown name (unknown_var = *global_var)
+                if (Token::simpleMatch(bodyToken->tokAt(-1), "*") && Token::simpleMatch(bodyToken->tokAt(-2), "=")) {
```

Comment:
in my opinion we need to be much more careful. If the global pointer is used then return false unless you can determine that there are no side effects. With this code it seems to me there will be false positives for lots of expressions. For instance `unknown_var = 1 + *global_var`. And lhs can be some expression also.. `unknown_buf[0] = *global_var`..